lib/internal/streams/pipe.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+const pipeline = require('internal/streams/pipeline');
+const Duplex = require('internal/streams/duplex');
+
+module.exports = function pipe(...streams) {
+  let cb;
+  let ret;
+
+  const r = pipeline(streams, function(err) {
+    if (cb) {
+      cb(err);
+    } else {
+      ret.destroy(err);
+    }
+  });
+  const w = streams[0];
+
+  ret = new Duplex({
+    writable: !!w?.writable,
+    readable: !!r?.readable,
+    objectMode: streams[0].readableObjectMode,
+    highWaterMark: 1
+  });
+
+  if (ret.writable) {
+    ret._write = function(chunk, encoding, callback) {
+      w.write(chunk, encoding, callback);
+    };
+
+    ret._final = function(chunk, encoding, callback) {
+      w.end(chunk, encoding, callback);
+    };
+  }
+
+  if (ret.readable) {
+    ret._read = function() {
+      r.resume();
+    };
+
+    r
+      .on('data', function(buf) {
```

Comment:
can we use the readable event instead?

Reply:
Any particular reason you prefer that?

Reply:
pause/resume has better compat with old streams I think?